synapse/storage/databases/main/metrics.py

```diff
@@ -411,7 +411,10 @@ def _generate_user_daily_visits(txn: LoggingTransaction) -> None:
                     ON u.user_id = udv.user_id AND u.device_id=udv.device_id
                     INNER JOIN users ON users.name=u.user_id
                     WHERE ? <= last_seen AND last_seen < ?
-                    AND udv.timestamp IS NULL AND users.is_guest=0
+                    AND udv.timestamp IS NULL
+                    AND u.user_agent !='sync-v3-proxy-'
+                    AND u.user_agent !='matrix-rust-sdk'
+                    AND users.is_guest=0
```

Comment:
Looking at the definition of the `user_ips` table it appears that there is one row per user/device/ip, and as such we only store *one* user agent per device. The upshot of this is that we won't necessarily pick up e.g. `Element X` user agents if its been clobbered by a sliding sync user agent, which I think will materially skew the stats here?

I wonder if we should instead be filtering out these user agents on insertion instead?

I'm still happy to merge this as is for a first cut if it'd be useful?

Reply:
Ah, good catch. No point in merging as is then. I'll edit to remove the UAs from insertion. 

Note the 'matrix-rust-sdk' entries generated from element x should reduce to zero following https://github.com/vector-im/element-x-ios/pull/1507

Reply:
FTR I think you want to do the filtering here:

https://github.com/matrix-org/synapse/blob/54a51ff6c1a1d9c7174e239acdd2dee7bed744f7/synapse/storage/databases/main/client_ips.py#L573-L604